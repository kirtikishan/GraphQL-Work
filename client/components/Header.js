import React, {Component} from 'react';
import BEMHelper from 'react-bem-helper';
import {Link} from 'react-router';

var header = new BEMHelper('header');
class Header extends Component {
    render(){
        return (
            <div {...header()}>
                <span {...header('home-link')}>
                    <Link to="/">Home</Link>
                </span>
                <span {...header('search-link')}>
                    <Link to="/search">Search</Link>
                </span>
            </div>
        );
    }
}

export default Header;