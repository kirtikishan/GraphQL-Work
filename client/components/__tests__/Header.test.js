import * as React from 'react';
import BEMHelper from 'react-bem-helper';
import {shallow} from 'enzyme';
import * as renderer from 'react-test-renderer';
import Header from '../Header';
import { Link } from 'react-router-dom';

describe('Header Component', () => {
    beforeEach(() => {
        const header = new BEMHelper('header');
    });

    it('should check the rendered snapshot', () => {
        const component = renderer.create(<Header />);
        expect(component.toJSON()).toMatchSnapshot();
    });

});
