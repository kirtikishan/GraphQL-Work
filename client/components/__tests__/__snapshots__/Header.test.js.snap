// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should check the rendered snapshot 1`] = `
<div
  className="header"
>
  <span
    className="header__home-link"
  >
    <a
      onClick={[Function]}
      style={Object {}}
    >
      Home
    </a>
  </span>
  <span
    className="header__search-link"
  >
    <a
      onClick={[Function]}
      style={Object {}}
    >
      Search
    </a>
  </span>
</div>
`;
